File: base/src/main/java/com/maubis/scarlet/base/database/daos/TagDao.java
package com.maubis.scarlet.base.database.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.maubis.scarlet.base.database.entities.Tag;

import java.util.List;

@Dao
public interface TagDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  long insertTag(Tag tag);

  @Delete
  void delete(Tag tag);

  @Query("SELECT count(*) FROM tag")
  int getCount();

  @Query("SELECT * FROM tag ORDER BY uid")
  List<Tag> getAll();

  @Query("SELECT * FROM tag WHERE uid = :uid LIMIT 1")
  Tag getByID(int uid);

  @Query("SELECT * FROM tag WHERE uuid = :uuid LIMIT 1")
  Tag getByUUID(String uuid);
}

File: base/src/main/java/com/maubis/scarlet/base/database/daos/TagDao.kt
package com.maubis.scarlet.base.database.daos

import androidx.room.*
import com.maubis.scarlet.base.database.entities.Tag

@Dao
interface TagDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTag(tag: Tag): Long

    @Delete
    fun delete(tag: Tag)

    @Query("SELECT count(*) FROM tag")
    fun getCount(): Int

    @Query("SELECT * FROM tag ORDER BY uid")
    fun getAll(): List<Tag>

    @Query("SELECT * FROM tag WHERE uid = :uid LIMIT 1")
    fun getByID(uid: Int): Tag

    @Query("SELECT * FROM tag WHERE uuid = :uuid LIMIT 1")
    fun getByUUID(uuid: String): Tag
}